src/hotspot/share/opto/parse1.cpp

```diff
@@ -734,6 +734,11 @@ void Parse::do_all_blocks() {
 
         // Leave behind an undisturbed copy of the map, for future merges.
         set_map(clone_map());
+
+        if (DoPartialEscapeAnalysis && block->is_loop_head()) {
+          PEAState& as = jvms()->alloc_state();
+          as.materialize_all();
```

Comment:
This patch LGTM.

Are we guaranteed to visit outer loop head before inner loop heads? For example:

```
Iterator it;
for (...)
  it.next();
  for (...)
    foo();
```

I want to make sure we materialize at the outer loop head, not the inner loop head.

Reply:
This patch doesn't materialize an object. It just converts its state from 'Virtual' to 'Escaped' in allocation state. no code change. 

yes, C2 Parse parses outer loop before inner loop because of RPO. RPO is the natural order as if we read a program.